layers/cip.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/google/gopacket"
+)
+
+const (
+	cipBasePacketLen int = 2
+)
+
+var (
+	ErrCIPDataTooSmall = errors.New("CIP packet data truncated")
+)
+
+type CIP struct {
+	BaseLayer
+	Response         bool     // false if request, true if response
+	ServiceID        byte     // The service specified for the request
+	ClassID          uint16   // request only
+	InstanceID       uint16   // request only
+	Status           byte     // Response only
```

Comment:
Potentially make this an enum, so that specific statuses can be printed in a human-readable way via .String()

Reply:
I'm nto sure about ServiceID/ClassID, but if they're a relatively small set of known-good values, maybe do this for them, too.

Reply:
Unfortunately, as I understand it, the ClassID and InstanceID fields are selected by the device manufacturer, and the only status I'm reasonably confident is consistent across manufacturers is 0x00 for success.